bundles/org.dataflowanalysis.analysis/src/org/dataflowanalysis/analysis/dsl/constraint/DSLNodeDestinationSelector.java

```diff
@@ -0,0 +1,99 @@
+package org.dataflowanalysis.analysis.dsl.constraint;
+
+import org.dataflowanalysis.analysis.dsl.AnalysisConstraint;
+import org.dataflowanalysis.analysis.dsl.variable.ConstraintVariable;
+import org.dataflowanalysis.analysis.dsl.selectors.CharacteristicsSelectorData;
+import org.dataflowanalysis.analysis.dsl.selectors.VertexCharacteristicsSelector;
+import org.dataflowanalysis.analysis.dsl.variable.ConstraintVariableReference;
+
+import java.util.List;
+
+/**
+ * Represents the DSL object of a node destination selector
+ */
+public class DSLNodeDestinationSelector {
+    private final AnalysisConstraint analysisConstraint;
+
+    /**
+     * Create a new DSL node destination selector object with the given analysis constraint
+     * @param analysisConstraint Given analysis constraint
+     */
+    public DSLNodeDestinationSelector(AnalysisConstraint analysisConstraint) {
+        this.analysisConstraint = analysisConstraint;
+    }
+
+    /**
+     * Match vertices that have the given node characteristic
+     * @param characteristicType Node characteristic type that must be present at the vertex
+     * @param characteristicValue Node characteristic value that must be present at the vertex
+     * @return DSL node selector to add more constraints
+     */
+    public DSLNodeDestinationSelector withCharacteristic(String characteristicType, String characteristicValue) {
```

Comment:
The data source is selected with labels, the destinations with characteristics, although both represent the same concept. What was our last state of naming here? If possible, we should try to unify the names, maybe something to discuss in the next meeting

Reply:
I agree, the terms are currently used interchangeably in the analysis

Reply:
Solution: Whenever possible, call it `label`, comprehensive refactoring for post 3.0.0.

Reply:
Moved to issue #178